src/Npgsql/Replication/PgOutput/PgOutputAsyncEnumerable.cs

```diff
@@ -112,7 +150,18 @@ await foreach (var xLogData in stream.WithCancellation(cancellationToken))
                 }
                 case BackendReplicationMessageCode.Relation:
                 {
-                    await buf.EnsureAsync(6);
+                    uint? transactionXid;
```

Comment:
nit: we could have a small local method utility which does this:

```c#
var transactionXid = EnsureInStreamingTraction(6);
```

Which internally checks inStreamingTransaction and returns 6 or 6+4 based on it.

Reply:
I was actually looking for a way to efficiently DRY that code but without success.
I'll have another look.

Reply:
Looking at this again, the most compact solution that came to my mind that still avoids introducing an additional async state machine for two lines of code was the following:

```csharp
await (inStreamingTransaction ? buf.EnsureAsync(10) : buf.EnsureAsync(6));
uint? transactionXid = inStreamingTransaction ? buf.ReadUInt32() : null;
```

While this is indeed shorter than the original below, it comes at the cost of checking `inStreamingTransaction` twice instead of once and (IMO) even being a bit harder to read.

```csharp
uint? transactionXid;
if (inStreamingTransaction)
{
    await buf.EnsureAsync(10);
    transactionXid = buf.ReadUInt32();
}
else
{
    await buf.EnsureAsync(6);
    transactionXid = null;
}
```

So in the end - if you don't mind - I'd be in favor of leaving it the (non-DRY) way it currently is.
The switch statement is pretty long in any case and you can always collapse the individual cases to gain back overview.

Reply:
Sure, this isn't very important. We can always revisit later if we want.